app/views/posts/show.html.erb

```diff
@@ -1,12 +1,12 @@
-<p>
-  <strong>User:</strong>
-  <%= @post.user_id %>
-</p>
+<h2><%= @post.content %></h2>
+Said by <%= @post.user.username %>
 
-<p>
-  <strong>Content:</strong>
-  <%= @post.content %>
-</p>
-
-<%= link_to 'Edit', edit_post_path(@post) %> |
-<%= link_to 'Back', posts_path %>
+<h3>This post has been favorited by: </h3>
+<% if @post.user_favorites.empty? %>
+  Nobody yet.
+<% else %>
+  <% @post.user_favorites.each do |favorite| %>
+    <%= favorite.username %><br />
+  <% end %>
+<% end %>
```

Comment:
Annie and I were talking about this kind of view logic a bit yesterday. I've never used a decorator, but I will look into it.